buildParts.js

```diff
@@ -0,0 +1,369 @@
+const { convertGraphQLToFQL } = require("./graphqlToFQLConverter");
+const {
+	capitalizeFirstLetter,
+	removeQuotes,
+	checkBool,
+	isNumeric
+} = require("./helper");
+const { operatorMap } = require("./constants");
+
+const functionCall = ' => ';
+const prefix = 'RE_';
+
+const getQueries = (data, queries) => {
+	if (!data)
+		return;
+
+	for (let index = 0; index < data.length; index++) {
+		const { all, any } = data[index];
+		if (all)
+			getQueries(all, queries);
+		else if (any)
+			getQueries(any, queries);
+		else {
+			const { source, target } = data[index];
+			if (source?.type?.toLowerCase() === 'fact')
+				queries.push(source.value);
+			if (target?.type?.toLowerCase() === 'fact')
+				queries.push(target.value);
+		}
+	}
+}
+
+const getCorrectOperator = (operator) => {
+	return operatorMap[operator.toLowerCase()] || operator;
+}
+
+const getTargetString = (operatorString, target) => {
+	const { type, value } = target;
+
+	if (type.toLowerCase() === 'fact')
+		return operatorString === '=='
+			? `${convertGraphQLToFQL(value)}`
+			: `(${convertGraphQLToFQL(value)})`;
+	else if (type.toLowerCase() === 'string')
+		return operatorString === '==' ? `"${value}"` : `("${value}")`
+
+	return value;
+}
+
+const createObjectMap = (data) => {
+
+	let topLevelMap = new Map();
+	let objectName;
+	let inputObject = {};
+	let factMap = new Map();
+	let conditionMap = new Map();
+
+	let oldName = Array.from(data.keys())[2].split('.');
+	oldName.pop();
+	oldName = oldName.join('.');
+
+	for (const [key, value] of data.entries()) {
+		const keys = key.split('.');
+		keys.pop();
+
+		const updatedAmount = keys.length;
+		const updatedKey = keys[updatedAmount-1];
+		const updatedName = keys.join('.');
+		const checkName = key.slice(0, updatedName.length);
+		const valueName = key.slice(updatedName.length + 1, key.length);
+
+		if (updatedKey !== undefined) {
+			// Create object name
+			if(updatedKey === 'source') {
+				if (valueName === 'value') {
+					objectName = createObjectName(value);
+				}
+				inputObject.source = {
+					...inputObject.source,
+					[valueName]: value
+				};
+			} else if (updatedKey === 'target') {
+				inputObject.target = {
+					...inputObject.target,
+					[valueName]: value
+				};
+			} else if (valueName === 'comparator') {
+				inputObject = {
+					...inputObject,
+					[valueName]: value
+				};
+			}
+
+			// Check if the whole object is set
+			// To create the sub objects
+			// And set it to the correct place
+			// TODO add check for old value of source/target
+			if (
+				inputObject.source?.type !== undefined
+				&& inputObject.source?.name !== undefined
+				&& inputObject.source?.value !== undefined
+				&& 'comparator' in inputObject
+				&& inputObject.target?.type !== undefined
+				&& inputObject.target?.value !== undefined
+			) {
```

Comment:
Replace `if (
				inputObject.source?.type !== undefined
				&& inputObject.source?.name !== undefined
				&& inputObject.source?.value !== undefined
				&& 'comparator' in inputObject
				&& inputObject.target?.type !== undefined
				&& inputObject.target?.value !== undefined
			) {`

with 
`if (
				inputObject.source?.type
				&& inputObject.source?.name
				&& inputObject.source?.value
				&& 'comparator' in inputObject
				&& inputObject.target?.type
				&& inputObject.target?.value
			) {
`